### ISSUE TITLE
redundantParens removes required parentheses around nested cast + subscript expressions

### ISSUE THREAD

**Issue by ChristopherRogers** [nicklockwood/SwiftFormat#386]
In 0.40.1, with the default options, one of the required sets of parentheses from the following code sample is removed.

Before:
```swift
let myDictionary: [AnyHashable: Any] = [:]
((myDictionary!["keyA"] as! [String: Any])["keyB"] as! [String: Any])["keyC"]
```

After:
```swift
let myDictionary: [AnyHashable: Any] = [:]
(myDictionary!["keyA"] as! [String: Any]["keyB"] as! [String: Any])["keyC"]
```

**Comment by nicklockwood** [comment]
@ChristopherRogers thanks for reporting. I've found the problem and will push a fix asap.

**Comment by nicklockwood** [comment]
@ChristopherRogers I've pushed a fix for this bug to the [develop](https://github.com/nicklockwood/SwiftFormat/tree/develop) branch if you'd like to try it.

**Comment by nicklockwood** [comment]
@ChristopherRogers fixed in 0.40.2.
